### DISCUSSION TITLE
Max Shard count, worker nodes, roadmap features

### DISCUSSION THREAD

**Discussion by anandsowm** [citusdata/citus#4832] [Q&A]
The citus v10 docs imply max shard count is 128 per worker node. Is this hard limit or likely to increase based on the h/w spec of the worker node? Also, docs indicate 100s of nodes are feasible in a cluster, is there a defined max number of nodes?
Where to look to know about roadmap feature items for future Citus releases ?

**Comment by marcocitus** [comment]
The shard count used by create_distributed_table is chosen based on the citus.shard_count GUC, which is 32 by default. This is the overall number of shards for the cluster. When scaling out, existing shards are moved to new nodes.

The shard count forms an upper-bound on the number of nodes. The maximum number of shards is currently around 64000.

It is possible to increase shard count using alter_distributed_table.
